cpp/include/cudf/detail/utilities/dremel.hpp

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/column/column.hpp>
+
+#include <rmm/device_uvector.hpp>
+
+namespace cudf::detail {
+
+/**
+ * @brief Device view for `dremel_data`.
+ *
+ * @see the `dremel_data` struct for more info.
+ */
+struct dremel_device_view {
+  size_type* offsets;
+  uint8_t* rep_levels;
+  uint8_t* def_levels;
+  size_type leaf_data_size;
+  uint8_t max_def_level;
+};
```

Comment:
This is mutable view instead. If we don't need to modify the data, let's make the pointers `type const*`.

Reply:
Otherwise, we may need to create two structs: view and mutable view.

Reply:
I can make the pointers const. Making the data members const too requires explicitly writing the constructor, but I guess that's not much extra work here.